test/functional/feature_blocksdir.py

```diff
@@ -17,18 +19,19 @@ def set_test_params(self):
 
     def run_test(self):
         self.stop_node(0)
-        node0path = os.path.join(self.options.tmpdir, "node0")
-        shutil.rmtree(node0path)
+        shutil.rmtree(self.nodes[0].datadir)
         initialize_datadir(self.options.tmpdir, 0)
         self.log.info("Starting with non exiting blocksdir ...")
-        self.assert_start_raises_init_error(0, ["-blocksdir="+self.options.tmpdir+ "/blocksdir"], "Specified blocks director")
-        os.mkdir(self.options.tmpdir+ "/blocksdir")
+        blocksdir_path = os.path.join(self.options.tmpdir, 'blocksdir')
+        self.nodes[0].assert_start_raises_init_error(["-blocksdir=" + blocksdir_path], re.escape('Error: Specified blocks directory "{}" does not exist.'.format(blocksdir_path)))
```

Comment:
instead of re.escape make this line be like:

```python
self.nodes[0].assert_start_raises_init_error(["-blocksdir=" + blocksdir_path], 'Error: Specified blocks directory "{!r}" does not exist.'.format(blocksdir_path))
```

Reply:
{!r} just calls substitutes in repr(blocksdir_path), it doesn't escape regexp symbols -- so {!r} would break with weird paths like "/src/bitcoin[test]/node0" because assert_start_raises_init_error would look for "/src/bitcoint/node0" or "/src/bitcoine/node0" or "/src/bitcoins/node0" and give spurious errors. PR seems fine as is to me.

Reply:
Makes sense